### PR TITLE
Add length_limit type options to keyword field mappers.

### PR DESCRIPTION
**Pull request by kovrus**
## Checklist

 - [ ] Added an entry in `CHANGES.txt` for user facing changes
 - [ ] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java [hunk 1]
```diff
@@ -199,6 +210,7 @@ public KeywordFieldMapper build(BuilderContext context) {
 
         private NamedAnalyzer normalizer = null;
         private boolean splitQueriesOnWhitespace;
+        private Integer lengthLimit;
```
Comment (kovrus) [inline server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java:213]: We could also keep it as primitive int with `Integer.MAX_VALUE` for the case that is represented as null now.

File: server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java [hunk 2]
```diff
@@ -420,7 +445,13 @@ protected String contentType() {
     @Override
     protected void doMerge(Mapper mergeWith, boolean updateAllTypes) {
         super.doMerge(mergeWith, updateAllTypes);
+        var mw = ((KeywordFieldMapper) mergeWith);
         this.ignoreAbove = ((KeywordFieldMapper) mergeWith).ignoreAbove;
```
Comment (mfussenegger) [inline server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java:449]:
```suggestion
        this.ignoreAbove = mw.ignoreAbove;
```

File: server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java [hunk 3]
```diff
@@ -672,10 +701,10 @@ private static boolean hasGaps(CachingTokenFilter stream) throws IOException {
             }
             return new PagedBytesIndexFieldData.Builder(fielddataMinFrequency, fielddataMaxFrequency, fielddataMinSegmentSize);
         }
-
     }
 
     private int positionIncrementGap;
+    private Integer lengthLimit;
```
Comment (mfussenegger) [inline server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java:707]: Is it necessary to have this property also on the `TextFieldMapper` ? Given that `VARCHAR(lengthLimit)` is mostly introduced for compatibility with SQL/PostgreSQL I think supporting it also for analyzed columns is not really necessary. If somebody adds an analyzer to a column, then it's already CrateDB dialect specific.

Comment (kovrus) [reply server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java:707]: hm shall we then forbid the declaration of the length limit on analyzed columns in the crate analyzer?

Comment (mfussenegger) [reply server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java:707]: Probably makes sense.

One other thing about analyzed columns is that they're tokenized, so whether the length would apply to the individual tokens or the raw input would also be a bit vague.

**Pull request merged**